evgen/array.hash.rb

```diff
@@ -0,0 +1,42 @@
+
+#2
+names.map(&:size).sum
+
+print names.each { |el| print el.length }
+
+count = 0
+names.each {|s| count += s.length}
+puts count
+
+#3
+names = names.map(&:downcase)
+
+#4
+hash = {}
+hash = names.each_with_object({}) do|element, object| object[element] = element.size
+end
+puts hash
+
+#5
+hash = {}
+names.map {|a| hash[a] = a.length}
+hash.key(hash.values.max)
+
+puts names.max_by(&:length)
+
+names.map(&:length).max
+
+#6
+longest_name = names.max_by(&:length)
+longest_name_index = names.index(longest_name)
+
+#7
+puts names.sort_by(&:length)
```

Comment:
You need to print out the part of the `names` array from `longest_name_index` till the last element